src/main/kotlin/exercise4/task1/BankAccount.kt

```diff
@@ -32,21 +32,51 @@ package exercise4.task1
  *      The initial balance for this constructor should always be set to 0.
  */
 
+open class BankAccount(private val accountNumber: String,
+                       private val accountHolderName: String,
+                       private var balance: Double = 0.0) {
+
+    open fun deposit(amount: Double): Boolean {
+        if (amount <= 0) {
+            println("Invalid deposit amount. The amount must be non-negative.\n")
+            return false
+        }
+        balance += amount
+        return true
+    }
+
+    open fun withdraw(amount: Double): Boolean {
+        if(balance < amount) {
+            println("There are not enough funds in the account to make the requested withdrawal.\n")
+        }
+        balance -= amount
+        return true
+    }
+
+    open fun getBalance(): Double = balance
```

Comment:
Other option is to change `balance` property visibility to `protected`

Reply:
It was one of the requirements to implement the `getBalance()` method. 

Also, the IDE recommends making the `balance` property private.